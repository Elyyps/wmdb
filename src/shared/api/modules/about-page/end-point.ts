import { generateAboutPageDummyData } from "./generate-dummy-data";

export const generateAboutPageData = generateAboutPageDummyData;
