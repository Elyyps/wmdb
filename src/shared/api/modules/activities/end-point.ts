import { generateActivitiesDummyData } from "./dummy-data";

export const generateActivitiesData = generateActivitiesDummyData;
