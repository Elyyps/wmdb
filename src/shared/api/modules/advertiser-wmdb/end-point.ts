import { generateAdvertiserDummyData } from "./dummy-data";
export const generateAdvertiserData = generateAdvertiserDummyData;
