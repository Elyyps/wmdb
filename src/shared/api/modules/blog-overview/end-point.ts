import { generateBlogOverviewDummyData, generateCards } from "./dummy-data";
import { ICompanyOutingCard } from "@app/api/core/company-outing-card";

export const generateBlogOverviewData = generateBlogOverviewDummyData;
const amountOfCards = 50;
export const getCardsWithPagination = (skip: number, take: number): ICompanyOutingCard[] => {
  let cards: ICompanyOutingCard[] = generateCards(amountOfCards);
  cards = cards.slice(skip, take + skip);

  return cards;
};
