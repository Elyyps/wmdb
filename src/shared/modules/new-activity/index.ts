export * from "./new-activity.component";
