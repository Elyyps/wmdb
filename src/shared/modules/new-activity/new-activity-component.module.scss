.new-activity {
  max-width: $container-max-width;
  margin: auto;
  &__columns {
    display: flex;
    flex-direction: row;
    margin: 0;
    justify-content: space-between;
    @media (max-width: $breakpoint-xsmall-max) {
      > div {
        margin-bottom: 16px;
      }
    }
  }
  &__cards {
    width: 45%;
  }
}
