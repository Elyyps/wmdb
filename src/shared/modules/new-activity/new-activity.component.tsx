import * as React from "react";
import styles from "./new-activity-component.module.scss";
import { OutingCardComponent } from "@app/core/outing-card";
import { INewActivity } from "@app/api/modules/new-activity/new-activity";
import { ModuleSectionComponent } from "@app/core/module-section";

export interface INewActivityComponentProps {
  newActivity: INewActivity;
}

const NewActivityComponent = (props: INewActivityComponentProps) => (
  <ModuleSectionComponent title={props.newActivity.title}>
    <div className={`${styles["new-activity__columns"]} uk-grid uk-child-width-1-2@m `}>
      {props.newActivity.cards.map((card, key) => (
        <div key={key}>
          <OutingCardComponent modify={"small"} {...card} />
        </div>
      ))}
    </div>
  </ModuleSectionComponent>
);

export { NewActivityComponent };
