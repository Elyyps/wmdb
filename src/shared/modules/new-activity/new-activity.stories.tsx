import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { NewActivityComponent } from "./new-activity.component";
import { generateNewActivityData } from "@app/api/modules/new-activity/end-point";

storiesOf("NewActivity", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <NewActivityComponent newActivity={generateNewActivityData()} />);
