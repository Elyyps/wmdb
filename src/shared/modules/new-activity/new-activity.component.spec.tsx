import React from "react";
import { shallow } from "enzyme";
import { NewActivityComponent } from "./new-activity.component";
import { generateNewActivityData } from "@app/api/modules/new-activity/end-point";

describe("[NewActivity]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<NewActivityComponent newActivityModule={generateNewActivityData()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
