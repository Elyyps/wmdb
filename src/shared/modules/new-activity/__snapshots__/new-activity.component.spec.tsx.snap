// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[NewActivity] should render component without crashing 1`] = `
<ModuleSectionComponent
  paddingBottom="72px"
  paddingTop="40px"
  pattern={true}
  title="Nieuwe activiteiten"
>
  <div
    className="new-activity__columns uk-grid uk-child-width-1-2@m "
  >
    <OutingCardComponent
      button={
        Object {
          "title": "Lees meer",
          "url": "/single-wmdb",
        }
      }
      categoriesId={
        Array [
          1,
        ]
      }
      content="Amsterdamse Uitjes: Speel het bekende spel ‘Escape Room’ nu als uniek stadsspel: Escape City. De sport van het aangaan van uitdagingen en opdrachten, het oplossen van raadsels en het ontcijferen van codes…"
      dataIcon="rooms.svg"
      image={
        Array [
          "https://picsum.photos/id/420/600/400",
          "https://picsum.photos/id/421/600/400",
          "https://picsum.photos/id/418/600/400",
          "https://picsum.photos/id/360/600/400",
        ]
      }
      key="0"
      maximumPersons={100}
      minimumPersons={10}
      modify="small"
      subtitle="Citygames in Egmond aan Zee"
      title="Escape City - Het ontsnappingsspel van Mokum #1"
    />
    <OutingCardComponent
      button={
        Object {
          "title": "Lees meer",
          "url": "/single-wmdb",
        }
      }
      categoriesId={
        Array [
          2,
        ]
      }
      content="TB Events: Met een flinke kater: verloren spullen, lippenstift op vreemde plaatsen, blauwe plekken... Net als  zelf real time...KEYWORD"
      dataIcon="rooms.svg"
      image={
        Array [
          "https://picsum.photos/id/115/300/200",
          "https://picsum.photos/id/116/300/200",
          "https://picsum.photos/id/117/300/200",
          "https://picsum.photos/id/118/300/200",
        ]
      }
      key="1"
      maximumPersons={100}
      minimumPersons={20}
      modify="small"
      subtitle="Citygames in Vijfhuizen"
      title="The Hangover #2x"
    />
  </div>
</ModuleSectionComponent>
`;
